eea/rdfmarshaller/browser/viewlets.py

```diff
@@ -0,0 +1,84 @@
+from Products.Five.browser.pagetemplatefile import ViewPageTemplateFile
+from plone.app.layout.viewlets.common import ViewletBase
+from plone.registry.interfaces import IRegistry
+from zope.component import getUtility
+import json
+
+
+REGISTRY_TYPE_LICENSES = "eea.rdfmarshaller.controlpanel.license." + \
+    "IPortalTypeLicenses.rdfmarshaller_type_licenses"
+
+REGISTRY_LICENSES = "eea.rdfmarshaller.controlpanel.license." + \
+    "ILicenses.rdfmarshaller_licenses"
+
+
+class LicenseViewlet(ViewletBase):
+    render = ViewPageTemplateFile('templates/license.pt')
+
+    @property
+    def license_text(self):
+        """ Return assigned license for portal_type of this context or None
+
+            Example:
+            https://github.com/theodi/open-data-licensing/blob/master/
+                    examples/simple-json-ld.json
+        """
+
+        registry = getUtility(IRegistry)
+        reg_types = registry.get(REGISTRY_TYPE_LICENSES, None)
+        reg_licenses = registry.get(REGISTRY_LICENSES, None)
+
+        text = None
+        if reg_licenses is not None and reg_types is not None:
+            if self.context.portal_type not in reg_types.keys():
+                return None  # No license assigned for this portal type
+
+            license_id = reg_types[self.context.portal_type]
+            licenses = [
+                x for x in reg_licenses if x['license_id'] == license_id]
+
+            if len(licenses) == 0:
+                return None  # No license details for this license id
+
+            license = licenses[0]
+            license_url = license.get("license_url", "")
+            license_title = license.get("license_id", "")
+
+            text = json.dumps(
```

Comment:
This is not using the eea.rdfmarshaller properly. This is work that isn't yet commited, still in work by me and Alec, so it's fine for now, but will need refactoring.